### PR TITLE
speech - integrate text-to-speech for chat items

### PR DESCRIPTION
**Pull request by bpasero**

### CODE REVIEW

File: src/vs/workbench/api/common/extHostSpeech.ts [hunk 1]
```diff
@@ -64,7 +68,11 @@ export class ExtHostSpeech implements ExtHostSpeechShape {
 		const cts = new CancellationTokenSource();
 		this.sessions.set(session, cts);
 
-		const textToSpeech = disposables.add(provider.provideTextToSpeechSession(cts.token));
+		const textToSpeech = await provider.provideTextToSpeechSession(cts.token);
```
Comment (amunger) [inline src/vs/workbench/api/common/extHostSpeech.ts:71]: does this still need to be added to disposables?

File: src/vs/workbench/api/common/extHostSpeech.ts [hunk 2]
```diff
@@ -36,7 +36,11 @@ export class ExtHostSpeech implements ExtHostSpeechShape {
 		const cts = new CancellationTokenSource();
 		this.sessions.set(session, cts);
 
-		const speechToTextSession = disposables.add(provider.provideSpeechToTextSession(cts.token, language ? { language } : undefined));
+		const speechToTextSession = await provider.provideSpeechToTextSession(cts.token, language ? { language } : undefined);
```
Comment (amunger) [inline src/vs/workbench/api/common/extHostSpeech.ts:39]: still add to disposables if defined?

Comment (bpasero) [reply src/vs/workbench/api/common/extHostSpeech.ts:39]: The session is no longer disposable, thats part of the PR.

File: src/vs/workbench/api/common/extHostSpeech.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/workbench/api/common/extHostSpeech.ts:71]: The session is no longer disposable, thats part of the PR.

**Review by amunger — APPROVED** [review]

File: src/vs/workbench/api/common/extHostSpeech.ts (hunk 2 above)
Comment (amunger) [reply src/vs/workbench/api/common/extHostSpeech.ts:39]: ah, I had checked the file's old version

**Pull request merged**